frontend/src/views/CreateProblemPage.tsx

```diff
@@ -0,0 +1,61 @@
+import React, { useState } from 'react';
+import styled from 'styled-components';
+import { useHistory } from 'react-router-dom';
+import { createProblem, Problem, ProblemIOType } from '../api/Problem';
+import { LargeText } from '../components/core/Text';
+import ErrorMessage from '../components/core/Error';
+import Loading from '../components/core/Loading';
+import ProblemDisplay from '../components/problem/ProblemDisplay';
+import Difficulty from '../api/Difficulty';
+import { TextLink } from '../components/core/Link';
+
+const Content = styled.div`
+  padding: 0 20%;
+`;
+
+function CreateProblemPage() {
+  const firstProblem = {
+    problemId: '',
+    name: 'Name',
+    description: 'Description',
+    difficulty: Difficulty.Easy,
+    testCases: [],
+    problemInputs: [],
+    outputType: ProblemIOType.Integer,
+  };
+
+  const history = useHistory();
+  const [problem, setProblem] = useState<Problem>(firstProblem);
+  const [error, setError] = useState('');
+  const [loading, setLoading] = useState(false);
+
+  const handleSubmit = (newProblem: Problem) => {
+    setLoading(true);
+    setError('');
+
+    createProblem(newProblem)
+      .then((res) => {
+        setProblem(res);
+        setLoading(false);
+
+        history.replace(`/problem/${res.problemId}`);
+      })
+      .catch((err) => {
+        setError(err.message);
+        setLoading(false);
+      });
+  };
+
+  return (
+    <Content>
+      <LargeText>Create Problem</LargeText>
+      <TextLink to="/problems/all">Back to all problems</TextLink>
```

Comment:
Sounds good, makes sense yeah

As for why I didn't include it in ProblemDisplay, I feel like logically the navigation link fits better in a Page.tsx file rather than in a problem display component

Reply:
That's a good point -- thanks for making the first change, and makes sense on not making the second change 👍